lib/common/base/data_wrapper.dart

```diff
@@ -0,0 +1,16 @@
+class DataWrapper<T> {
```

Comment:
to be honest I kinda disagree with DataWrapper concept, since Flutter `Future` is already covering `ConnectionStatus` and it also has `has data` or `has error` method which we can use for error handling. I think it wort to discuss more about this